### PR TITLE
Add support for MySQL 8.4

### PR DESCRIPTION
**Pull request by kolbe**
## A Pull Request should be associated with an Issue.

> We wish to have discussions in Issues. A single issue may be targeted by multiple PRs.
> If you're offering a new feature or fixing anything, we'd like to know beforehand in Issues,
> and potentially we'll be able to point development in a particular direction.
> Further notes in https://github.com/cashapp/spirit/blob/main/.github/CONTRIBUTING.md

Fixes https://github.com/cashapp/spirit/issues/269

### CODE REVIEW

File: scripts/manual_install_84.sh [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		mysql -u root -S "$datadir/mysql.sock" -e "SELECT 1" > /dev/null && return
+	done
+	printf %s\\n "MySQL check of $datadir failed"
+	return 1
+}
+
+
+deploy_replication() {
+	num_nodes=2
+	base_port=22334
```
Comment (morgo) [inline scripts/manual_install_84.sh]: This feature in dbdeployer is to use the MySQL version as the port. We don't need to follow it strictly, you can just set this to 3306.

Comment (kolbe) [reply scripts/manual_install_84.sh]: Doesn't dbdeployer start numbering ports from 22334 or something when you deploy a replication topology?

Comment (morgo) [reply scripts/manual_install_84.sh]: I forget the exact math, but it's something like 34 = 8.0.34. And the 22 = 8 + an offset of 14 or so for replica.

i.e. for version 8.4.0, the dbdeployer port is 8400. For replicas it adds an offset.

**Review by morgo — APPROVED** [review]
LGTM. Can you update this line in `README.md`:

> It only supports MySQL 8.0

**Pull request merged**